lib/system/zephyr/mbox.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2025 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	mbox.c
+ * @brief	Zephyr mbox primitives for libmetal.
+ *
+ */
+
+#include <metal/mbox.h>
+
+int __metal_mbox_send(metal_mbox_t *mbox, void *data, unsigned int len)
+{
+	metal_assert(mbox && mbox->m);
+	struct mbox_msg msg;
+	const int max_transfer_size_bytes = mbox_mtu_get_dt(mbox->m);
+
+	/* If data is NOT NULL, then pack message. */
+	if (data) {
```

Comment:
If I well remember  `data` can be not `NULL` for generic samples while the driver not support data so the `max_transfer_size_bytes != 0 ` should be tested first